Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -4931,6 +4932,201 @@ RefPtr<StyleValue> Parser::parse_content_value(Vector<ComponentValue> const& com
     return ContentStyleValue::create(StyleValueList::create(move(content_values), StyleValueList::Separator::Space), move(alt_text));
 }
 
+// https://w3c.github.io/csswg-drafts/css-display/#the-display-properties
+RefPtr<StyleValue> Parser::parse_display_value(Vector<ComponentValue> const& component_values)
+{
+    auto is_display_outside = [](ValueID identifier) -> bool {
+        switch (identifier) {
+        case ValueID::Block:
+        case ValueID::Inline:
+        case ValueID::RunIn:
+            return true;
+        default:
+            return false;
+        }
+    };
+    auto is_display_inside = [](ValueID identifier) -> bool {
+        switch (identifier) {
+        case ValueID::Flow:
+        case ValueID::FlowRoot:
+        case ValueID::Table:
+        case ValueID::Flex:
+        case ValueID::Grid:
+        case ValueID::Ruby:
+            return true;
+        default:
+            return false;
+        }
+    };
+
+    auto parse_inside = [](ValueID identifier) -> CSS::Display::Inside {
+        switch (identifier) {
+        case ValueID::Flow:
+            return CSS::Display::Inside::Flow;
+        case ValueID::FlowRoot:
+            return CSS::Display::Inside::FlowRoot;
+        case ValueID::Table:
+            return CSS::Display::Inside::Table;
+        case ValueID::Flex:
+            return CSS::Display::Inside::Flex;
+        case ValueID::Grid:
+            return CSS::Display::Inside::Grid;
+        case ValueID::Ruby:
+            return CSS::Display::Inside::Ruby;
+        default:
+            return {};
+        }
+    };
+    auto parse_outside = [](ValueID identifier) -> CSS::Display::Outside {
+        switch (identifier) {
+        case ValueID::Block:
+            return CSS::Display::Outside::Block;
+        case ValueID::Inline:
+            return CSS::Display::Outside::Inline;
+        case ValueID::RunIn:
+            return CSS::Display::Outside::RunIn;
+        default:
+            return {};
+        }
+    };
+
+    auto parse_single_component_display = [&](Vector<ComponentValue> const& component_values) -> Optional<CSS::Display> {
+        if (auto identifier = parse_identifier_value(component_values.first())) {
+            switch (identifier->to_identifier()) {
+
+                // display-outside
+            case CSS::ValueID::Block:
+                return CSS::Display { CSS::Display::Outside::Block, CSS::Display::Inside::Flow };
```

Comment:
I feel like these could make use of `CSS::Display::from_short()`. We already have the logic there for expanding a single display value into the full thing.

```suggestion
                return Display::from_short(Display::Short::Block);
```

Reply:
I didnt do it because i thought we could remove the extra `from_short` mapping completely later. I now think you are right because every place that changes the display would need to know how a value like `block` would expand into `block flow`. I will change it back for the single value mappings.

Reply:
Oh interesting. Yeah it would be fine to do this here except that there are other random places that want to do this conversion. It's nicer having it in one place.